### PR TITLE
eth-wallet: password flow based on the hash stored on the fs

### PR DESCRIPTION
**Pull request by Slava**
Versioning of the eth-wallet app is difficult so this update includes:
- new password flow based on hashing with bcrypt 
- "Generating..." ui for the wallets that are generated but not yet available (due to geth connectivity, or otherwise) #14991 #14989 #14961
- retry flow in case of geth not listening (retry every 2s until geth is available) #14826

### CODE REVIEW

File: app/ethWallet-geth.js [hunk 1]
```diff
@@ -293,6 +295,40 @@ ipcMain.on('eth-wallet-get-keys-path', (e) => {
...
+ipcMain.on('eth-wallet-get-compare-password-hash', (e, str, hash) => {
+  bcrypt.compare(str, hash, (err, res) => {
+    if (err) {
+      e.sender.send('eth-wallet-compare-password-hash', false)
+    } else {
+      e.sender.send('eth-wallet-compare-password-hash', res)
+    }
+  })
+})
+
```
Comment (ryanml) [inline app/ethWallet-geth.js]: In any case we are checking for `if (err)`, let's go ahead and output it via `console.error`

File: app/ethWallet-geth.js [hunk 2]
```diff
@@ -293,6 +295,40 @@ ipcMain.on('eth-wallet-get-keys-path', (e) => {
...
+      e.sender.send('eth-wallet-password-hash', null)
+    } else {
+      e.sender.send('eth-wallet-password-hash', hash)
+    }
+  })
+})
+
+ipcMain.on('eth-wallet-store-password-hash', (e, str) => {
+  bcrypt.genSalt(10, (err, salt) => {
+    if (err) {
```
Comment (ryanml) [inline app/ethWallet-geth.js]: In any case we are checking for `if (err)`, let's go ahead and output it via `console.error`

File: app/ethWallet-geth.js [hunk 3]
```diff
@@ -293,6 +295,46 @@ ipcMain.on('eth-wallet-get-keys-path', (e) => {
...
+    if (err || !hash) {
+      e.sender.send('eth-wallet-password-hash', null)
+    } else {
+      e.sender.send('eth-wallet-password-hash', hash)
+    }
+  })
+})
+
+ipcMain.on('eth-wallet-store-password-hash', (e, str) => {
+  bcrypt.genSalt(10, (err, salt) => {
```
Comment (diracdeltas) [inline app/ethWallet-geth.js]: 10 is kind of a small factor. did you benchmark this?

**Comment by Slava** [comment]
Cherry-picked to 0.23.x in https://github.com/brave/browser-laptop/commit/4df3efd87f12044f5e7117b6f1d6c776fc5d708f
Cherry-picked to 0.24.x in https://github.com/brave/browser-laptop/commit/c1d1fe92e331ca25f3b3d0d3ffa5368a8ff5ad56

**Pull request merged**